File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
num = int(input())
result = 0
for _ in range(num):
  letter = list(input())
  compare_arr = []
  isGroup = True
  length = len(letter) - 1
  for j in range(length):
    if(letter[j] != letter[j + 1] and letter[j] in compare_arr):
      isGroup = False
      break
    elif(letter[j] != letter[j + 1] and letter[j] not in compare_arr):
      compare_arr.append(letter[j])

    if(j == length - 1 and letter[j + 1] in compare_arr):
      isGroup = False

  if(isGroup):
    result += 1

print(result)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.c
#include <stdio.h>
#include <string.h>

int main(void) {  
   int num;
   scanf("%d", &num);

   int result = 0;

   for (int a = 0; a < num; a++) {
      char letter[101];
      int seen[26] = {0};
      int isGroupWord = 1;

      scanf("%s", letter);

      int length = strlen(letter);
      seen[letter[0] - 'a'] = 1;
      for (int i = 1; i < length; i++) {
         if (i > 0 && letter[i] != letter[i - 1]) {
            if (seen[letter[i] - 'a']) {
               isGroupWord = 0;
               break;
            }
         }
         seen[letter[i] - 'a'] = 1;
      }

      if (isGroupWord) {
         result++;
      }
   }

   printf("%d\n", result);

   return 0;
}